spec/trick_bag/io/gitignore_spec.rb

```diff
@@ -7,70 +7,45 @@
-        %w(root  a_dir/sub  .hidden_dir/.hidden_file  x/y/z/deep).each { |f| FileUtils.touch(f) }
-        File.write('.gitignore', ignore_spec.join("\n"))
-        yield
-      end
+  shared_examples_for 'testing a pair' do |ignore_spec, expected_result, filename|
+    let(:ignored_files)  { GI.list_ignored_files }
+    let(:in_ignore_list) { ignored_files.include?(filename) }
+    let(:context) do
+      "ignore_spec: #{ignore_spec}, ignored files: #{ignored_files}, filename: '#{filename}', expected result: #{expected_result}, got: #{in_ignore_list}"
     end
-  end
-
 
-  # Returns true on success, false on failure
-  def test_a_pair(ignore_spec, expected_to_be_in_ignore_list, filename)
-    success = :uninitialized
-    Dir.mktmpdir do |tmpdir|
-      Dir.chdir(tmpdir) do
-        File.write('.gitignore', ignore_spec.join("\n"))
-        Dir.mkdir('a_dir')       # for some but not all tests
-        Dir.mkdir('.hidden_dir') # for some but not all tests
-        FileUtils.touch(filename)      # for some but not all tests
-        ignored_files = GI.list_ignored_files
-        in_ignore_list = ignored_files.include?(filename)
-        success = (in_ignore_list == expected_to_be_in_ignore_list)
+    around do |example|
+      Dir.mktmpdir do |tmpdir|
+        Dir.chdir(tmpdir) do
+          File.write('.gitignore', ignore_spec.join("\n"))
+          Dir.mkdir('a_dir')        # for some but not all tests
+          Dir.mkdir('.hidden_dir')  # for some but not all tests
+          FileUtils.touch(filename) # for some but not all tests
+          example.run
+        end
       end
     end
-    success
-  end
-
-
-  def test_included_and_ignored(ignore_spec)
 
-    all_files = Dir.glob('**/*', File::FNM_DOTMATCH).reject do |filespec|
-      File.directory?(filespec)
-    end.sort
-    ignored_files = GI.list_ignored_files(ignore_spec).sort
-    included_files = GI.list_included_files(ignore_spec).sort
-
-    expect(included_files & ignored_files).to eq([])
-    expect((included_files | ignored_files).sort).to eq(all_files)
+    it { expect(in_ignore_list).to eq(expected_result), context }
   end
 
-
-  it 'should correctly handle a nonhidden file in the root directory' do
-
+  context 'handling a nonhidden file in the root directory', :aggregate_failures do
     test_inputs = [
-        [[],             false],
-        [['target'],     true],
-        [['**/*arge*'],  true],
-        [['*'],          true],
-        [['*arge*'],     true],
+      [[],             false],
+      [['target'],     true],
+      [['**/*arge*'],  true],
+      [['*'],          true],
+      [['*arge*'],     true],
     ]
 
-    failures = test_inputs.select do |(ignore_spec, expected_result)|
-      (test_a_pair(ignore_spec, expected_result, 'target') == false)
+    test_inputs.each do |(ignore_spec, expected_result)|
+      context do # without a context block the way that rspec expands the examples causes the paramters to overwrite each other
```

Comment:
Thank you for this comment! This is the kind of issue that, if I were writing code from scratch and omitted the inner context, might have had me stumped for hours without this explanation. Spelling issue: the word 'parameters' is missing its first 'e'.